src/views/studio/lib/studio-member-actions.js

```diff
@@ -187,6 +187,26 @@ const acceptInvitation = () => ((dispatch, getState) => new Promise((resolve, re
     });
 }));
 
+const transferHost = (password, newHostName, newHostId) =>
+    ((dispatch, getState) => new Promise((resolve, reject) => {
+        const state = getState();
+        const studioId = selectStudioId(state);
+        const token = selectToken(state);
+        newHostName = newHostName.trim();
+        api({
+            uri: `/studios/${studioId}/transfer/${newHostName}?password=${password}`,
```

Comment:
Is it always safe to embed a password right into the query parameters of a (PUT) request like this? If any part of the backend has a logging system which tracks requests by URL including non-GET, this code could inadvertently put users' passwords into a plain-text file accessible by any system admins—and possibly malicious hackers!

(This includes e.g. logging files created out of even sysadmins' control by the server providers. A mass breach of logs on those systems would leak Scratchers' passwords if PUT requests have their URLs logged.)

Disclaimer: this comment basically stolen from a comment by Chiroyce on codubee's profile, linking this line of code :P

Reply:
Is there any reason the password can't be inside the body of the request? Using query parameters for sending data in a request is also a little weird.

Reply:
I agree, it would be better to put the password in the body of the request.

Reply:
and is already in the process of being changed

Reply:
@colbygk 
> and is already in the process of being changed

Shouldn't this have been fixed before this was merged? Storing a password in the URL means that your password is perpetually stored in your browser history, which can be dangerous, especially if using Scratch on a shared device.